### PR TITLE
UCS/DEBUG: Allocate backtrace with mmap().

### LINKED ISSUE #1491: allocate backtrace with mmap rather than malloc
in case heap is corrupted

### PR DESCRIPTION
**Pull request by yosefe**
Fixes #1491

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1619/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3540/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1622/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1626/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3546/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@shamisp pls review

File: src/ucs/debug/debug.c [hunk 1]
```diff
@@ -122,6 +122,38 @@ khash_t(ucs_debug_symbol) ucs_debug_symbols_cache;
...
+        ucs_log_fatal_error("failed to allocate %zu bytes with mmap: %m", length);
+        return NULL;
+    }
+
+    return ptr;
+}
+
+static void ucs_debug_free_mem(void *ptr, size_t length)
+{
+    munmap(ptr, ucs_align_up_pow2(length, ucs_get_page_size()));
```
Comment (shamisp) [inline src/ucs/debug/debug.c]: Please add error check and print out the errno.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1641/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3561/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1644/ for details.

**Comment by yosefe** [comment]
@shamisp pls review again

**Pull request merged**